Tasks/Common/webdeployment-common/xmlvariablesubstitutionutility.ts

```diff
@@ -22,55 +22,56 @@ function getReplacableTokenFromTags(xmlNode, variableMap) {
                     tl.debug('Attribute value is in incorrect format ! ' + childNode.attrs[nodeAttribute]);
                     continue;
                 }
-                parameterSubValue[childNode.attrs[nodeAttribute].substring(indexOfReplaceToken, lastIndexOf_)] = variableMap[childNode.attrs['name']];
+                parameterSubValue[childNode.attrs[nodeAttribute].substring(indexOfReplaceToken, lastIndexOf_)] = variableMap[childNode.attrs['name']].replace(/"/g, "'");
             }
         }
     }
     return parameterSubValue;
 }
 
-
-async function substituteValueinParameterFile(parameterFilePath, parameterSubValue) {
+function substituteValueinParameterFile(parameterFilePath, parameterSubValue) {
 
     if(Object.keys(parameterSubValue).length === 0) {
         tl.debug('No substitution variables found for parameters.xml');
         return;  
     }
-
     var fileBuffer: Buffer = fs.readFileSync(parameterFilePath);
     var fileEncodeType = fileEncoding.detectFileEncoding(parameterFilePath, fileBuffer);
     var webConfigContent: string = fileBuffer.toString(fileEncodeType[0]);
     if(fileEncodeType[1]) {
         webConfigContent = webConfigContent.slice(1);
     }
     var xmlDocument;
+    const paramFileReplacableToken = 'PARAM_FILE_REPLACE_TOKEN';
+    var paramFileReplacableValues = {};
     try {
         xmlDocument = ltxdomutility.initializeDOM(webConfigContent);
         for(var xmlChildNode of xmlDocument.children) {
             if(!varUtility.isObject(xmlChildNode)) {
                 continue;
             }
             if(parameterSubValue[ xmlChildNode.attrs.name ]) {
-                xmlChildNode.attrs.defaultValue = parameterSubValue[ xmlChildNode.attrs.name ];
+                var paramFileReplacableTokenName = paramFileReplacableToken + '(' + xmlChildNode.attrs.name + ')';
+                xmlChildNode.attrs.defaultValue = paramFileReplacableTokenName;
+                tl.debug('Parameters file - Replacing value for name: ' + xmlChildNode.attrs.name + ' with : ' + paramFileReplacableTokenName);
+                paramFileReplacableValues[paramFileReplacableTokenName] = parameterSubValue[ xmlChildNode.attrs.name ];
             }
         }
     }
     catch(error) {
         tl.debug("Unable to parse parameter file : " + parameterFilePath + ' Error: ' + error);
         return;
     }
-
     var domContent = (fileEncodeType[1] ? '\uFEFF' : '') + ltxdomutility.getContentWithHeader(xmlDocument);
-
```

Comment:
why did you remove all these empty lines. this is a very big function and those separators were increasing the readability